### PR TITLE
Build package updates

### PR DESCRIPTION
**Pull request by hellkite500**
Some build/package updates

### CODE REVIEW

File: src/troute-network/setup.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def build_extensions(self):
     sources=[
              "troute/network/reservoirs/rfc/rfc.{}".format(ext),
              ],
-    include_dirs=[np.get_include()],
+    include_dirs=[np.get_include(),  "troute/network/"],
```
Comment (aaraney) [inline src/troute-network/setup.py:115]: Is this an issue if you try to install from a different directory than the dir that has `setup.py`?

File: src/troute-network/troute/nhd_io.py
Comment (aaraney) [inline src/troute-network/troute/nhd_io.py:1]: Is this supposed to be in this PR? Seems unrelated. Seems related to #616.

**Review by aaraney — COMMENTED** [review]
Looks good for the most part. I left a few minor questions and comments that should be easy to work through quickly. Thanks!

File: src/troute-network/troute/nhd_io.py
Comment (hellkite500) [reply src/troute-network/troute/nhd_io.py:1]: This was based on that branch, that is a shared commit 592afa754e09e0560b38ac3d99b41a7d616f9d7c so I can rebase this PR to clear it out if needed.

File: src/troute-network/setup.py (hunk 1 above)
Comment (hellkite500) [reply src/troute-network/setup.py:115]: I'm not sure about that, haha.

**Comment by shorvath-noaa** [comment]
@hellkite500 is this PR still being worked on or is it ready for review/merge?

**Comment by hellkite500** [comment]
@shorvath-noaa It should be good as is. Might need to do some testing based on @aaraney s comment and make some adjustments, but that can be done separately.

**Comment by shorvath-noaa** [comment]
@hellkite500 I got this to compile ok, but when I run t-route with RFC's on it still fails, so I'm not sure this fixes [Issue 620](https://github.com/NOAA-OWP/t-route/issues/620). This is the error I get:

```
At line 23 of file bind_rfc.f90
Fortran runtime error: Attempt to DEALLOCATE unallocated 'rfc_ptr'
At line 23 of file bind_rfc.f90
Fortran runtime error: Attempt to DEALLOCATE unallocated 'rfc_ptr'
Traceback (most recent call last):
  File "/home/sean.horvath/miniconda-template/miniconda3/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/sean.horvath/miniconda-template/miniconda3/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/sean.horvath/projects/t-route/src/troute-nwm/src/nwm_routing/__main__.py", line 2028, in <module>
    main_v04(v_args[1])
  File "/home/sean.horvath/projects/t-route/src/troute-nwm/src/nwm_routing/__main__.py", line 161, in main_v04
    run_results = nwm_route(
  File "/home/sean.horvath/projects/t-route/src/troute-nwm/src/nwm_routing/__main__.py", line 1074, in nwm_route
    results = compute_nhd_routing_v02(
  File "/home/sean.horvath/projects/t-route/src/troute-routing/troute/routing/compute.py", line 621, in compute_nhd_routing_v02
    results_subn[order] = parallel(jobs)
  File "/home/sean.horvath/miniconda-template/miniconda3/lib/python3.9/site-packages/joblib/parallel.py", line 1061, in __call__
    self.retrieve()
  File "/home/sean.horvath/miniconda-template/miniconda3/lib/python3.9/site-packages/joblib/parallel.py", line 938, in retrieve
    self._output.extend(job.get(timeout=self.timeout))
  File "/home/sean.horvath/miniconda-template/miniconda3/lib/python3.9/site-packages/joblib/_parallel_backends.py", line 542, in wrap_future_result
    return future.result(timeout=timeout)
  File "/home/sean.horvath/miniconda-template/miniconda3/lib/python3.9/concurrent/futures/_base.py", line 446, in result
    return self.__get_result()
  File "/home/sean.horvath/miniconda-template/miniconda3/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
    raise self._exception
joblib.externals.loky.process_executor.TerminatedWorkerError: A worker process managed by the executor was unexpectedly terminated. This could be caused by a segmentation fault while calling the function or by an excessive memory usage causing the Operating System to kill the worker.

The exit codes of the workers are {EXIT(2)}
```

**Comment by hellkite500** [comment]
@shorvath-noaa This fixes the build/packaging -- what you are seeing is a runtime error that is a separate issue, I think.

**Review by aaraney — APPROVED** [review]
Verified this installed correctly with `python3.8` on Debian slim bullseye.

**Review by shorvath-noaa — APPROVED** [review]
